mlflow/types/type_hints.py

```diff
@@ -0,0 +1,299 @@
+import inspect
+import logging
+from datetime import datetime
+from typing import Any, Literal, NamedTuple, Optional, Union, get_args, get_origin
+
+import pydantic
+
+from mlflow.exceptions import MlflowException
+from mlflow.types.schema import (
+    COLSPEC_TYPES,
+    AnyType,
+    Array,
+    ColSpec,
+    DataType,
+    Map,
+    Object,
+    Property,
+    Schema,
+)
+
+PYDANTIC_V1 = pydantic.VERSION.startswith("1.")
```

Comment:
```suggestion
PYDANTIC_V1_OR_OLDER = Version(pydantic.VERSION).major <= 1
```

Can we make this bool?